File: GOVI-Back/src/main/java/com/sncf/govi/service/model/Retournement.java
package com.sncf.govi.service.model;

import lombok.Builder;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Builder
@Data
public class Retournement {
    private CouleurEnum couleur;
    @Builder.Default
    private List<Mission> missionsArrivee = new ArrayList<>();
    @Builder.Default
    private List<Mission> missionsDepart = new ArrayList<>();
}

File: GOVI-Front/src/app/model/Retournement.ts
import { Mission } from './Mission';

export class Retournement {
  couleur: string;
  missionsArrivee: Mission[];
  missionsDepart: Mission[];
  constructor(
    couleur: string,
    missionsArrivee: Mission[],
    missionsDepart: Mission[]
  ) {
    this.couleur = couleur;
    this.missionsArrivee = missionsArrivee;
    this.missionsDepart = missionsDepart;
  }
}
